elastic/kibana
# Apm pages

## Summary

Converts APM pages from Angular to React

Fixes https://github.com/elastic/kibana/issues/111766

Overview:

![Screen_Shot_2021-10-04_at_16_24_18](https://user-images.githubusercontent.com/690/135810193-430f1310-7a0f-4839-86f9-1b733d9a1e8c.png)

Instances: Sorted by memory ascending

![Screen Shot 2021-10-04 at 16 27 56](https://user-images.githubusercontent.com/690/135810491-3f55149d-cd70-45d6-8dba-0da62362bd00.png)

Instances: Sorted by memory descending

![Screen Shot 2021-10-04 at 16 26 48](https://user-images.githubusercontent.com/690/135810409-fbf4ebf3-08d5-456f-b688-f41aff8ae06f.png)

Instances: Setup mode

![Screen_Shot_2021-10-04_at_16_28_23](https://user-images.githubusercontent.com/690/135810600-ed43c1c3-3ad7-4a64-aef3-9f0acd4dfe42.png)

Instance:

![Screen Shot 2021-10-04 at 16 29 28](https://user-images.githubusercontent.com/690/135810673-ec395587-0cce-457a-91e4-1f5a09f15e3d.png)

### Test

- Navigate to APM server
- Should land on the Overview index page with graphs
- Navigate to Instances
- Should land on APM server instance list
- Check sort/filter/paging
- Enter setup mode
- Should see "Monitored with metricbeat" and "Monitor another server with Metricbeat" to trigger slide out for configuration help
- Exit setup mode
- Navigate to a name
- Should see a single instance overview with graphs
- Verify it refreshes data if auto refreshing is on and looks the same in Angular mode

### Checklist

Delete any items that are not applicable to this PR.



### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/master/RISK_MATRIX.mdx) |


### For maintainers
